ui/src/app/settings/components/repos-list/repos-list.tsx

```diff
@@ -237,272 +233,357 @@ export class ReposList extends React.Component<RouteComponentProps<any>, {connec
-                        {projects => (
-                            <Form
-                                onSubmit={params => this.connectHTTPSRepo(params as NewHTTPSRepoParams)}
-                                getApi={api => (this.formApiHTTPS = api)}
-                                defaultValues={{type: 'git'}}
-                                validateError={(params: NewHTTPSRepoParams) => ({
-                                    url: (!params.url && 'Repository URL is required') || (this.credsTemplate && !this.isHTTPSUrl(params.url) && 'Not a valid HTTPS URL'),
-                                    name: params.type === 'helm' && !params.name && 'Name is required',
-                                    password: !params.password && params.username && 'Password is required if username is given.',
-                                    tlsClientCertKey: !params.tlsClientCertKey && params.tlsClientCertData && 'TLS client cert key is required if TLS client cert is given.'
-                                })}>
-                                {formApi => (
-                                    <form onSubmit={formApi.submitForm} role='form' className='repos-list width-control'>
-                                        <div className='white-box'>
-                                            <p>CONNECT REPO USING HTTPS</p>
-                                            <div className='argo-form-row'>
-                                                <FormField formApi={formApi} label='Type' field='type' component={FormSelect} componentProps={{options: ['git', 'helm']}} />
+                    {this.state.method === ConnectionMethod.SSH && (
+                        <DataLoader load={() => services.projects.list('items.metadata.name').then(projects => projects.map(proj => proj.metadata.name).sort())}>
+                            {projects => (
+                                <Form
+                                    onSubmit={params => this.connectSSHRepo(params as NewSSHRepoParams)}
+                                    getApi={api => (this.formApiSSH = api)}
+                                    defaultValues={{type: 'git'}}
+                                    validateError={(params: NewSSHRepoParams) => ({
+                                        url: !params.url && 'Repository URL is required'
+                                    })}>
+                                    {formApi => (
+                                        <form onSubmit={formApi.submitForm} role='form' className='repos-list width-control'>
+                                            <div className='white-box'>
+                                                <p>Choose your connection method:</p>
+                                                <DropDownMenu
+                                                    anchor={() => (
+                                                        <button className='argo-button argo-button--base'>
```

Comment:
One nit. I can think of one other place where a drop down menu is used in one of the sliders.  It's in the New App creation slider.  See the Source section.  It use a drop down menu with a different look.  Here, you are using a button style.  I think we should try to be consistent.  (Other places use an ellipses, but that isn't what we want here).